sophia/src/term/iri/_regex.rs

```diff
@@ -1,205 +1,27 @@
-            remove_dot_segments(&mut path);
-            query = iri_ref.query;
-        } else {
-            scheme = self.scheme;
-            if iri_ref.authority.is_some() {
-                authority = iri_ref.authority;
-                path = iri_ref.path.clone();
-                remove_dot_segments(&mut path);
-                query = iri_ref.query;
-            } else {
-                authority = self.authority;
-                if iri_ref.path.is_empty() {
-                    path = self.path.clone();
-                    query = iri_ref.query.or(self.query);
-                } else {
-                    if iri_ref.path[0] == "" {
-                        path = iri_ref.path.clone();
-                        remove_dot_segments(&mut path);
-                    } else {
-                        path = merge(&self, &iri_ref.path);
-                        remove_dot_segments(&mut path);
-                    }
-                    query = iri_ref.query;
-                }
-            }
-        }
-        fragment = iri_ref.fragment;
-        IriRefStructure {
-            scheme,
-            authority,
-            path,
-            query,
-            fragment,
-        }
-    }
-}
-
-impl fmt::Display for IriRefStructure<'_> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if let Some(scheme) = self.scheme {
-            write!(f, "{}:", scheme)?;
-        }
-        if let Some(authority) = self.authority {
-            write!(f, "//{}", authority)?;
-        }
-        if !self.path.is_empty() {
-            write!(f, "{}", self.path[0])?;
-            for p in &self.path[1..] {
-                write!(f, "/{}", p)?;
-            }
-        }
-        if let Some(query) = self.query {
-            write!(f, "?{}", query)?;
-        }
-        if let Some(fragment) = self.fragment {
-            write!(f, "#{}", fragment)?;
-        }
-        Ok(())
-    }
-}
-
-fn merge<'a>(base: &IriRefStructure<'a>, path: &[&'a str]) -> Vec<&'a str> {
-    let mut v = Vec::new();
-    if base.authority.is_some() && base.path.is_empty() {
-        v.push(""); // resulting path must have a leading '/'
-    }
-    v.extend(base.path.iter().take(base.path.len() - 1).cloned());
-    v.extend(path.iter().cloned());
-    v
-}
-
-fn remove_dot_segments(path: &mut Vec<&str>) {
-    if path.is_empty() {
-        return;
-    }
-    let mut i = 0;
-    let last = path[path.len() - 1];
-    if last == "." || last == ".." {
-        path.push("");
-    }
-    while i < path.len() {
-        if path[i] == "." {
-            path.remove(i);
-        } else if path[i] == ".." {
-            if i != 0 && (i != 1 || path[0] != "") {
-                path.remove(i - 1);
-                i -= 1;
-            }
-            path.remove(i);
-        } else {
-            i += 1;
-        }
-    }
-}
-
 lazy_static! {
-    static ref IRI_REGEX: Regex = Regex::new(r"(?x)^
+    /// Match an IRI reference either absolute or relative.
```

Comment:
```suggestion
    /// Match an absolute IRI reference
```
it does *not* match relative IRI references.

Note that, according to RFC3987, an IRI is always absolute. An **IRI reference**, on the other hand, can be absolute or relative (and an absolute IRI reference happens to be an IRI). That's why this regex is just called IRI_REGEX.

Reply:
Corrected documentation and added a note to documentation of `sophia::term::iri` about the naming convention.

I did not accept this as a commit to reduce absolute number of commits.